File: src/test/java/htsjdk/samtools/util/AbstractProgressLoggerTest.java
package htsjdk.samtools.util;

import htsjdk.HtsjdkTest;
import org.testng.Assert;
import org.testng.annotations.Test;

public class AbstractProgressLoggerTest extends HtsjdkTest {

    @Test
    public void padLeftPadsToTheGivenLength() {
        Assert.assertEquals(AbstractProgressLogger.pad("hello", 10), "     hello");
        Assert.assertEquals(AbstractProgressLogger.pad("hello", 6), " hello");
    }

    @Test
    public void padLeavesAStringOfExactlyTheGivenLengthUnchanged() {
        Assert.assertEquals(AbstractProgressLogger.pad("hello", 5), "hello");
    }

    @Test
    public void padLeavesAStringLongerThanTheGivenLengthUnchanged() {
        Assert.assertEquals(AbstractProgressLogger.pad("hello", 4), "hello");
    }

    @Test
    public void padLeavesTheStringUnchangedForANegativeLength() {
        Assert.assertEquals(AbstractProgressLogger.pad("hello", -1), "hello");
    }
}

File: src/test/scala/htsjdk/samtools/util/AbstractProgressLoggerTest.scala
package htsjdk.samtools.util

import htsjdk.UnitSpec

class AbstractProgressLoggerTest extends UnitSpec {

  "AbstractProgressLoggerTest.pad" should "pad the right amount of spaces in " in {
    Seq(
      ("hello", 10, "     hello"),
      ("hello", 6, " hello"),
      ("hello", 5, "hello"),
      ("hello", 4, "hello"),
      ("hello", -1, "hello")).foreach { case (in, len, expected) =>
      AbstractProgressLogger.pad(in, len) shouldEqual expected
    }
  }
}
